src/component_selector.jl

```diff
@@ -0,0 +1,124 @@
+# Most of this is implemented by wrapping the IS versions, replacing
+# InfrastructureSystemsComponent with Component and SystemData with System
+
+"""
+    get_components(selector, sys; filterby = nothing)
+Get the components of the `System` that make up the `ComponentSelector`.
+ - `filterby`: optional filter function to apply after evaluating the `ComponentSelector`
+"""
+get_components(selector::ComponentSelector, sys::System; filterby = nothing) =
+    IS.get_components(selector, sys; filterby = filterby)
+
+"""
+    get_components(filterby, selector, sys)
+Get the components of the `System` that make up the `ComponentSelector`.
+ - `filterby`: optional filter function to apply after evaluating the `ComponentSelector`
+"""
+get_components(
+    filterby::Union{Nothing, Function},
+    selector::ComponentSelector,
+    sys::System,
+) =
+    get_components(selector, sys; filterby = filterby)
```

Comment:
I think this would be slightly cleaner.
```
get_components(selector::ComponentSelector, sys::System) = IS.get_components(selector, sys)
get_components(filter by::Function, selector::ComponentSelector, sys::System) = get_components(selector, sys; filterby = filterby)
```

Reply:
`filter_func` instead of `filterby`? Even if the second is a better term, it's better to keep consistency in the docstrings.

Reply:
Related to our offline discussion, the fact that you can create the component selector with a filter function and then pass a separate filter function here adds complexity and possibly confusion to the user interface.

Reply:
Renamed this to `scope_limiter` to try to reduce confusion — see [this](https://github.com/NREL-Sienna/InfrastructureSystems.jl/pull/342#discussion_r1779178830) thread in the IS PR.

Reply:
`scope_limiter` is now no longer part of the user interface (see the IS PR).